### PR TITLE
Fix crash when preProcessScanlines allocs failed

### PR DESCRIPTION
**Pull request by NgaiShunChan**

### CODE REVIEW

File: lodepng.cpp [hunk 1]
```diff
@@ -5871,11 +5871,18 @@ unsigned lodepng_encode(unsigned char** out, size_t* outsize,
     if(!state->error) {
       state->error = lodepng_convert(converted, image, &info.color, &state->info_raw, w, h);
     }
-    if(!state->error) preProcessScanlines(&data, &datasize, converted, w, h, &info, &state->encoder);
+    if(!state->error)
```
Comment (lvandeve) [inline lodepng.cpp:5874]:
```suggestion
    if(!state->error) {
      state->error = preProcessScanlines(&data, &datasize, converted, w, h, &info, &state->encoder);
    }
```

File: lodepng.cpp [hunk 2]
```diff
@@ -5871,11 +5871,18 @@ unsigned lodepng_encode(unsigned char** out, size_t* outsize,
...
+    {
```
Comment (lvandeve) [inline lodepng.cpp:5875]:
```suggestion
    if(!state->error) {
```

File: lodepng.cpp [hunk 3]
```diff
@@ -5871,11 +5871,18 @@ unsigned lodepng_encode(unsigned char** out, size_t* outsize,
...
+      state->error = preProcessScanlines(&data, &datasize, converted, w, h, &info, &state->encoder);
+    }
     lodepng_free(converted);
     if(state->error) goto cleanup;
   }
-  else preProcessScanlines(&data, &datasize, image, w, h, &info, &state->encoder);
+  else
+  {
```
Comment (lvandeve) [inline lodepng.cpp:5882]:
```suggestion
  } else {
```

**Comment by lvandeve** [comment]
Thanks for the fix, looks good, I just left a few suggested changes to make the style of if/else braces match the rest of the code, please fix to make it match.

**Comment by NgaiShunChan** [comment]
Thanks for the review.

File: lodepng.cpp [hunk 4]
```diff
@@ -5871,11 +5871,15 @@ unsigned lodepng_encode(unsigned char** out, size_t* outsize,
     if(!state->error) {
       state->error = lodepng_convert(converted, image, &info.color, &state->info_raw, w, h);
     }
-    if(!state->error) preProcessScanlines(&data, &datasize, converted, w, h, &info, &state->encoder);
+    if(!state->error) {
+      state->error = preProcessS1canlines(&data, &datasize, converted, w, h, &info, &state->encoder);
```
Comment (lvandeve) [inline lodepng.cpp:5875]: Thanks for fixing! One more thing: the function name has a typo: preProcessS1canlines instead of preProcessScanlines

**Comment by lvandeve** [comment]
Thank you for the fix

**Pull request merged**